src/app_service/service/community_tokens/service.nim

```diff
@@ -303,3 +333,43 @@ QtObject:
     let data = ComputeDeployFeeArgs(ethCurrency: ethCurrency, fiatCurrency: fiatCurrency,
                                     errorCode: (if ethValue > balance: ComputeFeeErrorCode.Balance else: ComputeFeeErrorCode.Success))
     self.events.emit(SIGNAL_COMPUTE_DEPLOY_FEE, data)
+
+  proc fetchCommunityOwners*(self: Service, communityId: string, chainId: int, contractAddress: string) =
+    let arg = FetchCollectibleOwnersArg(
+      tptr: cast[ByteAddress](fetchCollectibleOwnersTaskArg),
+      vptr: cast[ByteAddress](self.vptr),
+      slot: "onCommunityTokenOwnersFetched",
+      chainId: chainId,
+      contractAddress: contractAddress,
+      communityId: communityId
+    )
+    self.threadpool.start(arg)
+
+  # get owners from cache
+  proc getCommunityTokenOwners*(self: Service, communityId: string, chainId: int, contractAddress: string): seq[CollectibleOwner] =
+    return self.tokenOwnersCache.getOrDefault((address: contractAddress, chainId: chainId))
+
+  proc onCommunityTokenOwnersFetched*(self:Service, response: string) {.slot.} =
+    let responseJson = response.parseJson()
```

Comment:
It's probably safer to put a `try/catch` around all of this, especially in case the returned JSON is not valid

Reply:
No need. I create this json in async call so it is always valid.